sessionctx/variable/session.go

```diff
@@ -801,3 +809,49 @@ type BatchSize struct {
 	// MaxChunkSize defines max row count of a Chunk during query execution.
 	MaxChunkSize int
 }
+
+const (
+	SlowLogPrefixStr     = "# "
+	SlowLogSpaceMarkStr  = ": "
+	SlowLogSQLSuffixStr  = ";"
+	SlowLogTimeStr       = "Time"
+	SlowLogTxnStartTSStr = "Txn_start_ts"
+	SlowLogUserStr       = "User"
+	SlowLogConnIDStr     = "Conn_ID"
+	SlowLogQueryTimeStr  = "Query_time"
+	SlowLogDBStr         = "DB"
+	SlowLogIsInternalStr = "Is_internal"
+	SlowLogIndexNamesStr = "Index_names"
+	SlowLogQuerySQLStr   = "Query" // use for slow log table, slow log will not print this field name but print sql directly.
+)
+
+// SlowLogFormat uses for formatting slow log.
+func (s *SessionVars) SlowLogFormat(txnTS uint64, costTime time.Duration, execDetail execdetails.ExecDetails, indexNames string, sql string) string {
+	var buf bytes.Buffer
+	execDetailStr := execDetail.String()
+	buf.WriteString(fmt.Sprintf("# %s: %v\n", SlowLogTxnStartTSStr, txnTS))
```

Comment:
maybe we'd better not use `Sprintf` for simple concat

https://mp.weixin.qq.com/s?__biz=MzA4Mzg2NzE5MA==&mid=2648672554&idx=1&sn=aee385754a3124676b5f31685af2c1b8&chksm=87c4ca73b0b3436519f563298bb3dd63b35e93513e870ebd27ec6fd693e3af21a394d08c2e1e&mpshare=1&scene=1&srcid=02230yAStbFOqJEUoR9qZKih&pass_ticket=EnmeSikPb6RX%2BsCN0F0A1%2FVN7AET5%2FUbXIKJnVqOj18%3D#rd
https://medium.com/@robiplus/how-to-efficiently-concatenate-strings-in-go-6eb1a875573f

Reply:
done.